pkg/bgpls/bgpls-nlri.go

```diff
@@ -460,9 +460,8 @@ func (ls *NLRI) GetUnreservedLinkBandwidthKbps() []uint64 {
 		if tlv.Type != 1091 {
 			continue
 		}
-		for p := 0; p < len(tlv.Value); {
-			unResrved = append(unResrved, uint64(math.Float32frombits(binary.BigEndian.Uint32(tlv.Value[p:p+4])) * 8 / 1000))
-			p += 4
+		for i, p := 0, 0; p < len(tlv.Value); i, p = i+1, p+4 {
+			unResrved[i] = uint64(math.Float32frombits(binary.BigEndian.Uint32(tlv.Value[p:p+4])) * 8 / 1000)
```

Comment:
the issue exists in the old code too, in case tlv length is wrong, unchecked p:p+4 will cause a panic. Do you want to add extra length check just to be on the safe side?

Reply:
Yeah sounds good, I added commit that checks the length, in the case that TLV is not expected length, function will return the initial value of all zeros so as to not introduce new error handling but does log the error so this event should not be too horribly obfuscated.